internal/controller/finbackupconfig_controller_test.go

```diff
@@ -0,0 +1,299 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+func createControllerPod(name, namespace string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: "test-sa",
+			Containers: []corev1.Container{
+				{
+					Name:  "manager",
+					Image: "test:latest",
+				},
+			},
+		},
+	}
+}
+
+func findEnvVar(envVars []corev1.EnvVar, name string) *corev1.EnvVar {
+	for i := range envVars {
+		if envVars[i].Name == name {
+			return &envVars[i]
+		}
+	}
+	return nil
+}
+
+var _ = Describe("FinBackupConfig Controller", func() {
+	var reconciler *FinBackupConfigReconciler
+	var ctx context.Context
+
+	BeforeEach(func() {
+		ctx = context.Background()
+
+		reconciler = NewFinBackupConfigReconciler(
+			k8sClient,
+			scheme.Scheme,
+			"",
+		)
+		reconciler.finCephClusterID = namespace
+	})
+
+	AfterEach(func() {
+		Expect(os.Unsetenv("POD_NAME")).To(Succeed())
+		Expect(os.Unsetenv("POD_NAMESPACE")).To(Succeed())
+		Expect(os.Unsetenv("CREATE_FINBACKUP_JOB_SERVICE_ACCOUNT")).To(Succeed())
+	})
+
+	Describe("CronJob creation", func() {
+		It("should create a CronJob when FBC is created", func() {
+			// Description:
+			//   Verify a CronJob is correctly generated from a FinBackupConfig.
+			//
+			// Arrange:
+			//   - Prepare an RBD StorageClass, PVC/PV, controller Pod, and env vars.
+			//   - Create a FinBackupConfig.
+			//
+			// Act:
+			//   - Call Reconcile().
+			//
+			// Assert:
+			//   - No error is returned.
+			//   - CronJob's schedule, owner, JobTemplate and container settings are as expected.
+
+			By("creating RBD StorageClass")
+			sc := NewRBDStorageClass("test", namespace, rbdPoolName)
+			err := k8sClient.Create(ctx, sc)
+			Expect(err).NotTo(HaveOccurred())
+			defer func() { _ = k8sClient.Delete(ctx, sc) }()
+
+			By("creating PVC and PV using the StorageClass")
+			pvc, pv := NewPVCAndPV(sc, namespace, "test-pvc", "test-pv", rbdImageName)
+			err = k8sClient.Create(ctx, pvc)
+			Expect(err).NotTo(HaveOccurred())
+			err = k8sClient.Create(ctx, pv)
+			Expect(err).NotTo(HaveOccurred())
+			defer func() {
+				DeletePVCAndPV(ctx, pvc.Namespace, pvc.Name)
+			}()
+
+			By("creating controller Pod")
+			pod := createControllerPod("test-controller", namespace)
+			err = k8sClient.Create(ctx, pod)
+			Expect(err).NotTo(HaveOccurred())
+			defer func() { _ = k8sClient.Delete(ctx, pod) }()
+
+			By("setting up environment variables")
+			Expect(os.Setenv("POD_NAME", pod.Name)).To(Succeed())
+			Expect(os.Setenv("POD_NAMESPACE", pod.Namespace)).To(Succeed())
+			Expect(os.Setenv("CREATE_FINBACKUP_JOB_SERVICE_ACCOUNT", "test-sa")).To(Succeed())
+
+			By("creating FinBackupConfig")
+			fbc := &finv1.FinBackupConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-fbc",
+					Namespace: namespace,
```

Comment:
Let's use a different namespace from that of the controller Pod to make things more realistic.